src/inc/piece/state/shifting_state.h

```diff
@@ -0,0 +1,23 @@
+#ifndef JARED_ALL_TETRIS_SHIFTING_STATE
+#define JARED_ALL_TETRIS_SHIFTING_STATE
+
+#include "tetris_piece_state.h"
+
+class ShiftingState : public TetrisPieceState
```

Comment:
Instead of the TetrisPieceState abstract class handling the left or right shift, I decided to make it a state on it's own. This is more flexible, as the different states following a shift can now be explicitly identified